### PR TITLE
EVG-3040 save spot instance IDs

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: model/host/host.go [hunk 1]
```diff
@@ -26,6 +26,9 @@ type Host struct {
 	Distro   distro.Distro `bson:"distro" json:"distro"`
 	Provider string        `bson:"host_type" json:"host_type"`
 
+	// secondary (external) identifier for the host
+	ExtIdentifier string `bson:"ext_identifier" json:"ext_identifier"`
```
Comment (tychoish) [inline model/host/host.go]: maybe name this external rather than Ext

File: cloud/ec2_client.go [hunk 2]
```diff
@@ -46,6 +47,12 @@ type AWSClient interface {
 	// DescribeSpotInstanceRequests is a wrapper for ec2.DescribeSpotInstanceRequests.
 	DescribeSpotInstanceRequests(context.Context, *ec2.DescribeSpotInstanceRequestsInput) (*ec2.DescribeSpotInstanceRequestsOutput, error)
 
+	// DescribeSpotRequestsAndSave is a wrapper for DescribeSpotInstanceRequests that also saves instance IDs to the db
+	DescribeSpotRequestsAndSave(context.Context, []*host.Host) (*ec2.DescribeSpotInstanceRequestsOutput, error)
```
Comment (tychoish) [inline cloud/ec2_client.go]: With the addition of this, we should probably not have the method on line 48 and use this everywhere?

**Comment by tychoish** [comment]
it might be good to write some extra tests.

File: model/host/host.go (hunk 1 above)
Comment (john-m-liu) [reply model/host/host.go]: done

File: cloud/ec2_client.go (hunk 2 above)
Comment (john-m-liu) [reply cloud/ec2_client.go]: done

**Pull request merged**